Userland/Libraries/LibGLSL/AST.h

```diff
@@ -0,0 +1,751 @@
+        : Expression(parent, start, end, filename)
+    {
+    }
+};
+
+class VariableDeclaration : public VariableOrParameterDeclaration {
+public:
+    virtual ~VariableDeclaration() override = default;
+    virtual StringView class_name() const override { return "VariableDeclaration"sv; }
+    virtual void dump(FILE* = stdout, size_t indent = 0) const override;
+
+    VariableDeclaration(ASTNode const* parent, Optional<Position> start, Optional<Position> end, DeprecatedString const& filename)
+        : VariableOrParameterDeclaration(parent, start, end, filename)
+    {
+    }
+
+    virtual bool is_variable_declaration() const override { return true; }
+
+    Expression const* initial_value() const { return m_initial_value; }
+    void set_initial_value(RefPtr<Expression const>&& initial_value) { m_initial_value = move(initial_value); }
+
+private:
+    RefPtr<Expression const> m_initial_value;
+};
+
+class Identifier : public Expression {
+public:
+    virtual ~Identifier() override = default;
+    virtual StringView class_name() const override { return "Identifier"sv; }
+    virtual void dump(FILE* = stdout, size_t indent = 0) const override;
+
+    Identifier(ASTNode const* parent, Optional<Position> start, Optional<Position> end, DeprecatedString const& filename, StringView name)
+        : Expression(parent, start, end, filename)
+        , m_name(name)
+    {
+    }
+    Identifier(ASTNode const* parent, Optional<Position> start, Optional<Position> end, DeprecatedString const& filename)
+        : Identifier(parent, start, end, filename, {})
+    {
+    }
+
+    virtual bool is_identifier() const override { return true; }
+
+    StringView name() const { return m_name; }
+    void set_name(StringView&& name) { m_name = move(name); }
+
+private:
+    StringView m_name;
+};
+
+class Name : public Expression {
+public:
+    virtual ~Name() override = default;
+    virtual StringView class_name() const override { return "Name"sv; }
+    virtual void dump(FILE* = stdout, size_t indent = 0) const override;
+    virtual bool is_name() const override { return true; }
+    virtual bool is_sized() const { return false; }
+
+    Name(ASTNode const* parent, Optional<Position> start, Optional<Position> end, DeprecatedString const& filename)
+        : Expression(parent, start, end, filename)
+    {
+    }
+    virtual StringView full_name() const;
+
+    Identifier const* name() const { return m_name.ptr(); }
+    void set_name(RefPtr<Identifier const>&& name) { m_name = move(name); }
+    Vector<NonnullRefPtr<Identifier const>> const& scope() const { return m_scope; }
+    void set_scope(Vector<NonnullRefPtr<Identifier const>> scope) { m_scope = move(scope); }
+    void add_to_scope(NonnullRefPtr<Identifier const>&& part) { m_scope.append(move(part)); }
+
+private:
+    RefPtr<Identifier const> m_name;
+    Vector<NonnullRefPtr<Identifier const>> m_scope;
+    mutable Optional<DeprecatedString> m_full_name;
```

Comment:
Could you try not to introduce any new usages of `DeprecatedString` and use `String` instead?